### DISCUSSION TITLE
After upgrading to Webpack 5, css-loader fails in runtime with TypeError: Failed to construct 'URL': Invalid URL

### DISCUSSION THREAD

**Discussion by viljark** [webpack/webpack#14193] [Q&A] [answered]
After migrating our app to Webpack 5, it builds nicely, but in runtime I get

`TypeError: Failed to construct 'URL': Invalid URL`

And the failing line is here:

```
var ___CSS_LOADER_URL_IMPORT_0___ = new URL(/* asset import */ __webpack_require__("../../node_modules/@mdi/font/fonts/materialdesignicons-webfont.eot?v=5.9.55"), __webpack_require__.b);
```

The first argument resolves nicely to `"/assets/materialdesignicons-webfont.eot?53f53f50a1b2033979f58d724a02a7e5"`, but the second argument is `undefined`.

Looking at the source code of css-loader, the second argument comes from `import.meta.url`.

How come is this undefined and does someone have any suggestions what could be wrong or is this a bug with css-loader?

**Comment by alexander-akait** [comment]
Something in your plugins break it, please provide steps to reproduce

**Comment by viljark** [comment, marked as answer]
I fixed it by specifying `target: 'web'` manually, seems like Webpack defaults did not work for me.

**Reply by alexander-akait** [reply]
Do you have the problem with dev server? `web` is default and should work, maybe you override this option in CLI arguments or in custom plugin

**Reply by viljark** [reply]
Dev server worked fine without providing target. I only had issues with production build.
There is only one plugin different in the production build, I'll check if this plugin is causing the issues (probably tomorrow).

**Reply by alexander-akait** [reply]
Feel free to feedback

**Reply by viljark** [reply]
The plugin did not cause it, so I will continue with `target: 'web'` solution for now.
